java/src/main/native/CMakeLists.txt

```diff
@@ -87,6 +88,10 @@ if(CUDF_USE_PER_THREAD_DEFAULT_STREAM)
   add_compile_definitions(CUDA_API_PER_THREAD_DEFAULT_STREAM CUDF_USE_PER_THREAD_DEFAULT_STREAM)
 endif()
 
+if(CUDF_JNI_ENABLE_PROFILING)
+  target_compile_definitions(cudfjni PUBLIC CUDF_JNI_ENABLE_PROFILING)
```

Comment:
@rapidsai/rapids-cmake-codeowners Can someone look at this CMake? Do I need a value here or something like `"CUDF_JNI_ENABLE_PROFILING=1"`? I don't feel super confident in this.

Reply:
I'm not a CMake expert, but this looks correct to me.

Reply:
Based on the build failure, I think this should be `add_compile_definitions` like the code just above.

Reply:
I am going to push a fix for this shortly. I think this CMake file has some issues. We _should_ prefer `target_compile_definitions` to keep the scope limited, but `add_library(cudfjni)` occurs lower in the script. All `target_compile_definitions` must happen after the library is defined. This is also an issue for `USE_NVTX`, which I was using as an example. I am guessing that we have not recently tested the build with that option turned off.

Reply:
both `CUDF_JNI_ENABLE_PROFILING` and `CUDF_JNI_ENABLE_PROFILING=1` are valid. It depends on if you want to define a name or a name with value.

```cpp
#define CUDF_JNI_ENABLE_PROFILING 
// versus
#define CUDF_JNI_ENABLE_PROFILING 1
```

Reply:
But yes this needs to go after the construction of the target `cudfjni`